OSSM-4815: Document HA for a mesh (#96010)
openshift/openshift-docs: modules/ossm-configuring-istio-ha-replicacount.adoc

// This procedure is used in the following assembly:
// * service-mesh-docs-main/install/ossm-installing-service-mesh.adoc

:_mod-docs-content-type: PROCEDURE
[id="ossm-configuring-istio-ha-replicacount_{context}"]
= Configuring Istio HA by using replica count

> Configure the {istio} control plane in High Availability (HA) mode to prevent a single point of failure, and ensure continuous mesh operation even if one of the `istiod` pods fails. The replica count defines a fixed number of {istio} control plane pods that can operate. Use replica count for mesh environments in which the number of workloads does not scale.

Author: done

.Prerequisites

* You are logged in to the {ocp-product-title} web console as a user with the `cluster-admin` role.

* You installed the {SMProductName} Operator.

> * You deployed the {istio} resource.

Reviewer: ```suggestion
* You have deployed the `{istio}` resource.
```

.Procedure

> . Obtain the name of the {istio} resource by running the following command:

Reviewer: ```suggestion
. Obtain the name of the `{istio}` resource by running the following command:
```

+
[source,terminal]
----
$ oc get istio -n istio-sytem
----
+
.Example output
[source,terminal]
----
NAME      REVISIONS   READY   IN USE   ACTIVE REVISION   STATUS    VERSION   AGE
default   1           1       0        default           Healthy   v1.24.6   24m
----
+

> The name of the {istio} resource is `default`.

Reviewer: ```suggestion
The name of the `{istio}` resource is `default`.
```

Author: done

> . Update the {istio} custom resource by adding the `autoscaleEnabled` and `replicaCount` parameters by running the following command:

Reviewer: ```suggestion
. Update the `{istio}` custom resource (CR) by adding the `autoscaleEnabled` and `replicaCount` parameters by running the following command:
```

+
[source,terminal]
----
$ oc patch istio default -n istio-system --type merge -p '
spec:
  values:
    pilot:
      autoscaleEnabled: false <1> 
      replicaCount: 2 <2>
'
----

> <1> Disables autoscaling and ensures that the number of replicas remains fixed.
> <2> Defines the number of {istio} control plane replicas. To support HA, there must be at least two replicas.

Reviewer: ```suggestion
<1> Specifies a setting that disables autoscaling and ensures a fixed number of replicas.
<2> Specifies the number of {istio} control plane replicas. To support HA, there must be at least two replicas.
```

.Verification

> . Verify the status of the {Istio} control pods by running the following command:

Reviewer: ```suggestion
* Verify the status of the {Istio} control pods by running the following command:
```

Author: done
